source/_topics/packages.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: page
+title: "Packages"
+description: "Describes all there is to know about configuration packages in Home Assistant."
+date: 2017-01-10 20:00 +0200
+sidebar: false
+comments: false
+sharing: true
+footer: true
+---
+
+## {% linkable_title About %}
+
+Packages in Home Assistant provides a way to bundle different component's configuration together. We were already introduced to the two configuration styles (specifying platforms entries together or individually) on the [adding devices](/getting-started/devices) page. Both of these configuration methods require you to create the component key in the main `configuration.yaml` file. With packages we have a way to include different components, or parts of configuration using any of the `!include` directives introduced in [splitting the configuration](/topics/splitting_configuration).
+
+Packages are configured under the core `homeassistant/packages` in the configuration and take the format of a packages name (no spaces, all lower case) followed by a dictionary with the package config. For example, package `pack_1` would be created as:
+
+```yaml
+homeassistant:
+  ...
+  packages: 
+    pack_1:
+      ...package config here...
+```
+
+The package configuration can include: `switch`, `light`, `automation`, `groups` or the majority of the Home Assistant components. 
+
+It can be specified inline, or in a seperate yaml file using `!include`
+
+Inline example, main `configuration.yaml`:
+
+```yaml
+homeassistant:
+  ...
+  packages: 
+    pack_1:
+      switch:
+        - platform: rest
+          ...
+      light:
+        - platform: rpi
+          ...
+```
+
+Include example, main `configuration.yaml`:
+```yaml
+homeassistant:
+  ...
+  packages: 
+    pack_1: !include my_package.yaml
+```
+The file `my_package.yaml` contains the "top-level" configuration:
+```
+switch:
+  - platform: rest
+    ...
+light:
+  - platform: rpi
+    ...
+```
+
+There are some rules for packages that will be merged:
+
+1. Component names may only use the basic form (e.g. `switch` and `switch 1` or `switch aa` is not accepted)
+2. Platform based components (`light`, `switch`, etc) can always be merged
+3. Components where entities are identified by a key that will represent the entity_id (`{key: config}`) need to have unique 'keys' between packages and the main configuration file (`input_boolean`, `group`)
```

Comment:
I think this needs an example of what not to do. It was hard to understand the first time I read this.